MetaLamp/lending-pool/client/src/Components/App.purs

```diff
@@ -0,0 +1,203 @@
+module Components.App where
+
+import Prelude
+
+import Business.Aave as Aave
+import Capability.Contract (class Contract, ContractId(..))
+import Capability.Delay (class Delay)
+import Capability.LogMessages (class LogMessages, logError, logInfo)
+import Components.AmountForm as AmountForm
+import Control.Monad.Except (lift, runExceptT, throwError)
+import Data.Array (mapWithIndex)
+import Data.Bifunctor (bimap, lmap)
+import Data.BigInteger (BigInteger, fromInt)
+import Data.Either (Either(..), either)
+import Data.Foldable (find)
+import Data.Json.JsonTuple (JsonTuple(..))
+import Data.Json.JsonUUID (JsonUUID(..))
+import Data.Maybe (Maybe(..))
+import Data.Symbol (SProxy(..))
+import Data.Tuple (Tuple(..))
+import Data.UUID (toString) as UUID
+import Halogen as H
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Network.RemoteData (RemoteData(..), fromEither)
+import Network.RemoteData as RD
+import Plutus.Contracts.Core (Reserve(..))
+import Plutus.Contracts.Endpoints (BorrowParams(..), DepositParams(..), RepayParams(..), WithdrawParams(..))
+import Plutus.PAB.Webserver.Types (ContractInstanceClientState(..))
+import Plutus.V1.Ledger.Crypto (PubKeyHash)
+import Plutus.V1.Ledger.Value (AssetClass(..), TokenName(..), Value)
+import PlutusTx.AssocMap as Map
+import View.FundsTable (fundsTable)
+import View.RemoteDataState (remoteDataState)
+import View.ReserveInfo (reserveInfo)
+import Wallet.Emulator.Wallet (Wallet(..))
+import Wallet.Types (ContractInstanceId(..))
+
+type State =
+  { contractId :: RemoteData String ContractId,
+    walletPubKey :: RemoteData String PubKeyHash,
+    userFunds :: RemoteData String Value,
+    reserves :: RemoteData String (Map.Map AssetClass Reserve),
+    lastStatus :: RemoteData String String }
```

Comment:
Is this Wallet Contract Instance State? Maybe we need to name this more precisely? 
P.S. in future we could use `(Wallet Contract Instance State, Wallet Contract Instance State)` or `[Wallet Contract Instance State]` for `State`